### ISSUE TITLE
[bug] Static alternative url generation if pagination is enabled on a multi-lang project

### ISSUE THREAD

**Issue by XaviTorello** [lektor/lektor#216]
A project with some alternative content defined:

```
$ cat blabla.lektorproject
...
[alternatives.one]
name = One
url_prefix = /one/
locale = XX
primary = yes

[alternatives.two]
name = Two
url_prefix = /two/
locale = YY
...
```

If pagination is enabled for a model:

```
$ cat models/wtf.ini
...
[pagination]
enabled = yes
per_page = 10
...
```

, and the model have some alternative content

```
$ cat content/wtf.lr
bla = blabla

$ cat content/wtf+two.lr
bla = albalb
```

, trying to fetch an alternative url for this content/model

```
$ cat templates/wtf.html
This is an <a href="{{ '.'|url(alt='two') }}">Alternative</a>
```

**returns a wrong path** -> ever returns the primary alternative

```
This is an <a href="/one/wtf">Alternative</a>
```

**instead of:**  //expected result

```
This is an <a href="/two/wtf">Alternative</a>
```

Reproduced with different _slug for each alternative, and without it.

**Comment by XaviTorello** [comment]
**Temporal workaround**

If pagination can be avoided for this model, handle the same funct with an specific macro keeping pagination inactive for this model

**Comment by XaviTorello** [comment]
**Workaround if pagination must be activated**

use

```
This is an <a href="{{ site.get(this._id, alt=lang).url_to('.') }}">Alternative</a>
```

instead of

```
This is an <a href="{{ '.'|url(alt=lang) }}">Alternative</a>
```

The fist one will return the correct URL taking care about the alternative, and the second one the primary path.

It's not a clean workaround but it works...

**Comment by nixjdm** [comment]
Fixed by #594
